compiler/rustc_codegen_ssa/src/errors.rs

```diff
@@ -494,7 +494,11 @@ impl<G: EmissionGuarantee> Diagnostic<'_, G> for LinkingFailed<'_> {
                             Err(_) => false,
                         };
                         let mut arg = dir.into_os_string();
-                        arg.push("/{");
+                        arg.push("/");
+                        let needs_braces = 2 <= rlibs.len();
```

Comment:
never really seen a constant on the left of a comparison.. This should be clearer?

```suggestion
                        let needs_braces = rlibs.len() >= 2;
```

Reply:
That's habit (not [Yoda](https://en.wikipedia.org/wiki/Yoda_conditions)). I always write comparisons in the order of the comparison for 2 reasons:
- `3 <= x && x <= 8` is more readable than `x >= 3 && x <= 8` in my opinion
- it always works, while the `subject <= object` rule doesn't work when there's 2 subjects like `j >= i`

I didn't think there would be a convention here. Happy to change it.